util/execdetails/execdetails.go

```diff
@@ -555,9 +555,22 @@ type RuntimeStatsColl struct {
 }
 
 // NewRuntimeStatsColl creates new executor collector.
-func NewRuntimeStatsColl() *RuntimeStatsColl {
-	return &RuntimeStatsColl{rootStats: make(map[int]*RootRuntimeStats),
-		copStats: make(map[int]*CopRuntimeStats)}
+// Reuse the object to reduce allocation when *RuntimeStatsColl is not nil.
+func NewRuntimeStatsColl(reuse *RuntimeStatsColl) *RuntimeStatsColl {
+	if reuse != nil {
+		for k := range reuse.rootStats {
```

Comment:
Reuse map is cheap than create a new map object.
Go compiler optimize this cleanup code pattern to a  `clearmap()` function.